### DISCUSSION TITLE
Merge with LocalStorage?

### DISCUSSION THREAD

**Discussion by MichelJansson** [Blazored/SessionStorage#60] [Ideas]
Thanks for all the great Blazored packages!

I include both LocalStorage and SessionStorage packages in my projects and I can imagine others do too, so asked myself - why do we need two?

As I understand it, there is virtually no difference between window.localStorage and window.sessionStorage except for the lifetimes of the data stored in each (they both expose a Storage object). As such, the code between the these packages should be almost identical.

If merging or creating a new "BrowserStorage" package is something that sound like a good idea, I would be happy to take a stab at it.

Thanks

**Comment by chrissainty** [comment]
This is a fair point and something I've thought about several times. I've even got a repo created for it. I've never actually done it as we'd lose all the downloads and stars the current repos and NuGet packages have. That counts for a lot when people are looking for a package.

But perhaps we could look at having the combined repo alongside the existing packages until the downloads and stars grow to a decent level? It's a tricky one I feel.

**Reply by MichelJansson** [reply]
I fully understand your point.

I see at least two scenarios for the repos with different benefits/drawbacks. When thinking about the pros/cons, I am assuming the combined repo would be the main repo for further development.

However the repos is dealt with, I think both local and session packages could be dependent on the combined, and have little or no code them selves.

EDIT:
Perhaps don't change the repos at all, just introduce session storage interfaces into the local storage package.

-----

1. Rename LocalStorage -> BrowserStorage + new LocalStorage repo for backwards compatibility.
   Pros:
     - LocalStorage is the package with the most GH stars/activity, and all that 
 would be transferred to the updated combined repo.
     - NuGet downloads would remain for LocalStorage
     - Git history could be kept
     - Automatic exposure for the combined repo

   Cons:
     - A little of a dance to get done.
     - No NuGet downloads for combined repo.

2. New combined, others kept as is
   Pros:
      - Easy

   Cons:
      - No stats for the main repo.
      - Potentially wasted stats, when archiving the other repos (as opposed to get them transferred to the main repo)

3. Mix: Start with 2. and the later swap them as in 1.
    Pros/Cons, basically all from both.
